pkg/pillar/hypervisor/containerd.go

```diff
@@ -174,15 +174,16 @@ func (ctx ctrdContext) GetDomsCPUMem() (map[string]types.DomainMetric, error) {
 	}
 
 	for _, id := range ids {
-		var usedMem, availMem uint32
+		var usedMem, availMem, totalMem uint32
 		var usedMemPerc float64
 		var cpuTotal uint64
 
 		if metric, err := containerd.CtrGetContainerMetrics(id); err == nil {
 			usedMem = uint32(roundFromBytesToMbytes(metric.Memory.Usage.Usage))
-			availMem = uint32(roundFromBytesToMbytes(metric.Memory.Usage.Max))
-			if availMem != 0 {
-				usedMemPerc = float64(100 * float32(usedMem) / float32(availMem))
+			totalMem = uint32(roundFromBytesToMbytes(metric.Memory.HierarchicalMemoryLimit))
+			availMem = totalMem - usedMem
```

Comment:
Can this become negative?

Reply:
I would be *really* surprised if it does (since these #s are basically pulled right from cgroups stats and I don't think that can happen there). That that -- better safe than sorry so we may as well check for that.

Reply:
@rvs @eriknordmark I've updated the changes, please take a look.